Automattic/woocommerce-payments
# Update the subscriptions empty state screen code to only handle the "not-connected" case

Fixes #6900

#### Changes proposed in this Pull Request

In WooPayments v3.4 we introduced empty state screens for the WooCommerce Subscriptions admin list table. see https://github.com/Automattic/woocommerce-payments/pull/3403

As part of that PR there were 2 variations of that screen.

1. If the merchant had not onboarded to WooPayments it would include a [button for them to onboard](https://user-images.githubusercontent.com/8490476/142810806-73e70740-03bb-4d32-a108-5ffc86def5a9.png).
2. If the merchant had onboarded, it would include default screen which mentioned that subscriptions will be listed here and a link to create a product.

The second screen has now been introduced to Subscriptions core as the default empty state screen and so no longer needs to be handled by WooPayments. 

This PR updates WooPayments so it only now handles the case where the store hasn't onboarded and so should be encouraged to onboard. It can fallback to the default screen in when it is connected.

#### Testing instructions

1. Make sure the WC Subscriptions plugin is disabled.
2. Make sure you're running `fix/wcs-4549` of subscriptions-core or the latest once that branch is merged. 
2. Delete all your subscriptions.
   - _If you'd prefer to keep them, you can do a bulk post type update and restore it after testing._
3. Go to the **WooCommerce > Subscriptions** screen. 
   - If you've onboarded to WooPayments you should see the default grayscale screen. 
   - If you haven't onboarded to WooPayments you should see the blue image with a onboarding button. 
6. Purchase a subscription. 
7. Go to the subscriptions admin list table.
    - You should see the subscription you just purchased listed.
8. Filter the list table by searching for an dummy string - we want to test 0 results.
    - You should just see the "No subscriptions found." text.


| Connected | Not connected |
|--------|--------|
| <img width="518" alt="Screenshot 2023-08-02 at 9 46 56 am" src="https://github.com/Automattic/woocommerce-payments/assets/8490476/61321645-b81c-412b-9ef1-51807cfa668a"> | <img width="518" alt="Screenshot 2023-08-03 at 1 40 59 pm" src="https://github.com/Automattic/woocommerce-payments/assets/8490476/24c46712-92f3-4da7-8488-9d5d9cfefbf2"> | 

-------------------


**Post merge**
